Homework 2 completed

daxdue opened the pull request.

Commit:
Homework 1 done.

Commit:
Some issues fixed:
- Data provider's constants moved to TestsDataProvider class
- Added required modifiers for constants
- Removed unnecessary blank lines
- Added version properties for some dependencies in pom.xml

Commit:
Added calculator jar file, fixed calculator dependency definition in pom

Commit:
Homework 2 completed.

Commit:
Homework 2 completed.

komyagin commented on pom.xml:
```diff
@@ -102,9 +102,9 @@
+            <!--<activation>
                 <activeByDefault>true</activeByDefault>
-            </activation>
+            </activation>-->
```
Why do need this comment here?

komyagin commented on src/test/java/ru/training/at/hw2/exercise1/FirstExerciseTest.java:
```diff
@@ -25,6 +26,7 @@ public void setUp() {
         webDriver = new ChromeDriver();
         webDriver.manage().window().maximize();
         webDriver.navigate().to("https://jdi-testing.github.io/jdi-light/index.html");
```
URL must be moved to the constant

Commit:
Removed hw2 folder

komyagin committed:
Merge pull request #3 from daxdue/homework_1

Fixed issue with hw2 folders

Commit:
Fixed issues:
- Same behaviour moved to abstract class
- URL moved to constant
- Locators refactored
- Webdrivermanager added.

Commit:
Fixed issues:
- Same behaviour moved to abstract class
- URL moved to constant
- Locators refactored
- Webdrivermanager added.

komyagin commented on src/test/java/ru/training/at/hw2/exercise1/FirstExerciseTest.java:
```diff
@@ -0,0 +1,101 @@
+     */
+    @Test(priority = 6)
+    public void iframeExistingTest() {
+        WebElement frame = webDriver.findElement(By.xpath("//iframe[@id='frame']"));
```
Maybe just use By.id?

komyagin commented on src/test/java/ru/training/at/hw2/exercise1/FirstExerciseTest.java:
```diff
@@ -0,0 +1,101 @@
+    @Test(priority = 5)
+    public void textsAmountTest() {
+        List<WebElement> benefitTexts = webDriver
+                .findElements(By.cssSelector("span.benefit-txt:nth-child(2)"));
```
There is more suitable locator

komyagin commented on src/test/java/ru/training/at/hw2/exercise1/FirstExerciseTest.java:
```diff
@@ -0,0 +1,101 @@
+     */
+    @Test(priority = 7)
+    public void frameButtonExistingTest() {
+        WebElement frame = webDriver.findElement(By.xpath("//iframe[@id='frame']"));
```
You can move webElement to a variable if you use it more than one time. Maybe just use By.id?

komyagin commented on src/test/java/ru/training/at/hw2/exercise1/FirstExerciseTest.java:
```diff
@@ -0,0 +1,101 @@
+    @Test(priority = 8)
+    public void sidebarElementsExistingTest() {
+        List<WebElement> sideBarItems = webDriver
+                .findElements(By.cssSelector("ul.sidebar-menu > li > a"));
```
There is more suitable locator

komyagin commented on src/test/java/ru/training/at/hw2/exercise2/SecondExerciseTest.java:
```diff
@@ -0,0 +1,79 @@
+    @Test(priority = 4)
+    public void selectCheckboxesTest() {
+        WebElement waterCheckBox = webDriver.findElement(By.cssSelector(
+                "label.label-checkbox:nth-child(1) > input"));
```
There is a more suitable locator

komyagin commented on src/test/java/ru/training/at/hw2/exercise2/SecondExerciseTest.java:
```diff
@@ -0,0 +1,79 @@
+        waterCheckBox.click();
+        Assert.assertTrue(waterCheckBox.isSelected());
+        WebElement windCheckBox = webDriver.findElement(By.cssSelector(
+                "label.label-checkbox:nth-child(3) > input"
```
There is a more suitable locator

komyagin commented on src/test/java/ru/training/at/hw2/exercise2/SecondExerciseTest.java:
```diff
@@ -0,0 +1,79 @@
+    @Test(priority = 5)
+    public void selectRadioTest() {
+        WebElement selenRadio = webDriver.findElement(By.cssSelector(
+                "label.label-radio:nth-child(4) > input"
```
There is a more suitable locator

Commit:
Fixed issues:
- Locators refactored
- Added common method to find input elements
- Added common method to find li elements

Commit:
Fixed issues:
- Locators refactored
- Added common method to find input elements
- Added common method to find li elements

Commit:
Merge remote-tracking branch 'origin/homework_2' into homework_2
